raven/base.py

```diff
@@ -636,7 +636,13 @@ def capture(self, event_type, data=None, date=None, time_spent=None,
             **kwargs)
 
         # should this event be sampled?
-        if self._random.random() < self.sample_rate:
+        sample_rate = self.sample_rate
+        try:
+            sample_rate = float(extra['sample_rate'])
```

Comment:
if you did that this code would need to live in the logging abstraction, and not here.

the 'extra' bit is 'extra context to send with the event', not to be confused with 'configuration for this event'